### PR TITLE
Initial GitHub Actions Setup

### LINKED ISSUE #45: Migrate to GitHub Actions instead of TravisCI
Using GitHub actions for CI instead of TravisCI is beneficial as it allows us to eliminate a third party service we are reliant on. There are also more use cases for GitHub Actions in automation that we should explore.

### PR DESCRIPTION
**Pull request by blaketarter**
## Description

Initial Setup of GitHub Actions to replace TravisCI.

Currently it:
- installs
- runs tests
    - lint
    - types
    - unit
    - visual
    - e2e
- uploads visual test results if the tests fail
- runs coverage and uploads results to codecov
- builds

Closes #45 

## 🔎 Reviewer Checklist

> Checked off by the PR **Reviewers**

### Required

> These always need to be checked

- [ ] Merge destination is correct
- [ ] Code is correct as understood and conforms to quality standards
- [ ] Tests have been added where appropriate (unit, visual, end-to-end)
- [ ] Acceptance Criteria have been met

---

>### Roles & Responsibilities
>
>#### 👨‍💻 Assignee
>
>- Initiator of this PR (be sure to set in GitHub UI)
>- Addresses feedback and change requests
>- Merges PR once approved (usually deletes branch unless `develop` or `release`)
>
>#### 👩‍💻 Reviewer
>
>- Invited to review PR by **Assignee** (via GitHub UI)
>- Is expected to complete a review and address followup

### CODE REVIEW

**Comment by roblafeve** [comment]
https://github.com/cremalab/crema-app-web/blob/c91b85fbf5da0e26f052443f336c1e2512285adb/package.json#L81

Can now be (I think): `"!src/**/*.stories.{ts,tsx}", `

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +0,0 @@
...
-    packages:
-      # Ubuntu 16+ does not install this dependency by default, so we need to install it ourselves
-      - libgconf-2-4
-install:
-  - npm ci
-  - npm install -g codecov
-script:
-  - npm run test:ci
-after_success:
-  - npm run test:unit:coverage:ci
```
Comment (roblafeve) [inline .travis.yml:15]: Just a passing thought, but I wonder if we should add a bunch of these to docs for those wanting to integrate with various CIs 🤔

File: package.json [hunk 2]
```diff
@@ -57,9 +57,8 @@
     "test:playground:build": "build-storybook",
     "test:playground:ci": "start-storybook -p 5000 -s public --ci",
     "test:types": "tsc",
-    "test:unit": "npm test",
+    "test:unit": "npm run test --",
```
Comment (roblafeve) [inline package.json:60]: When do you need the `--` and when don't you. Perpetually confused.

File: package.json [hunk 3]
```diff
@@ -78,10 +77,10 @@
   "jest": {
     "collectCoverageFrom": [
       "!node_modules",
-      "!src/**/(stories).{ts,tsx}",
+      "!src/**/*.stories.{ts,tsx}",
```
Comment (roblafeve) [inline package.json:80]: 👌

File: package.json [hunk 4]
```diff
@@ -78,10 +77,10 @@
...
       "!src/index.tsx",
       "!src/serviceWorker.ts",
-      "!src/**/(test).{ts,tsx}",
+      "!src/**/*.test.{ts,tsx}",
```
Comment (roblafeve) [inline package.json:83]: 👌

**Review by roblafeve — DISMISSED** [review]
Great work! Thanks for getting this converted 🎉 

(**Let's squash and merge to keep this clean**)

File: .github/workflows/CI.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
+name: CI
+
+on: [push]
+
+jobs:
+  test:
```
Comment (roblafeve) [inline .github/workflows/CI.yml]: Can jobs be given names like `Test`?

Comment (roblafeve) [reply .github/workflows/CI.yml]: It would look prettier here:

image

File: .travis.yml (hunk 1 above)
Comment (blaketarter) [reply .travis.yml:15]: That's a really good idea!

File: .github/workflows/CI.yml (hunk 5 above)
Comment (blaketarter) [reply .github/workflows/CI.yml]: I think so

File: package.json (hunk 2 above)
Comment (blaketarter) [reply package.json:60]: Me too, I had to play around with it until the coverage tests worked again.

File: .travis.yml (hunk 1 above)
Comment (roblafeve) [reply .travis.yml:15]: Cookbook for each CI we encounter 👍

File: .github/workflows/CI.yml (hunk 5 above)
Comment (roblafeve) [reply .github/workflows/CI.yml]: Mind trying?

**Comment by roblafeve** [comment]
And after all this, remember to `Squash and merge` if you don't mind 🙏

**Comment by roblafeve** [comment]
I should have specified "From the UI". That way conversation and commits are linked and we preserve the work. No biggie!

File: .github/workflows/CI.yml (hunk 5 above)
Comment (blaketarter) [reply .github/workflows/CI.yml]: Worked!

Screen Shot 2019-11-26 at 2 36 20 PM

**Comment by roblafeve** [comment]
FYI—when a branch is protected, you have to select the processes that need to pass before "All Checks have passed" is true. In the repo settings.

**Review by roblafeve — APPROVED** [review]
:shipit:

**Pull request merged**